src/test/java/com/farion/onlinebookstore/controller/CategoryControllerTest.java

```diff
@@ -0,0 +1,125 @@
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.post;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.jsonPath;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+
+import com.farion.onlinebookstore.dto.book.BookDto;
+import com.farion.onlinebookstore.dto.category.CategoryDto;
+import com.farion.onlinebookstore.dto.category.CreateCategoryRequestDto;
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import java.util.List;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.http.MediaType;
+import org.springframework.security.test.context.support.WithMockUser;
+import org.springframework.test.context.jdbc.Sql;
+import org.springframework.test.web.servlet.MockMvc;
+import org.springframework.test.web.servlet.MvcResult;
+import org.springframework.test.web.servlet.request.MockMvcRequestBuilders;
+import org.springframework.test.web.servlet.setup.MockMvcBuilders;
+import org.springframework.web.context.WebApplicationContext;
+import org.testcontainers.shaded.org.apache.commons.lang3.builder.EqualsBuilder;
+
+@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT)
+public class CategoryControllerTest {
+    private static MockMvc mockMvc;
+    @Autowired
+    private ObjectMapper objectMapper;
+
+    @BeforeAll
+    static void beforeAll(@Autowired WebApplicationContext context) {
+        mockMvc = MockMvcBuilders
+                .webAppContextSetup(context)
+                .apply(springSecurity())
+                .build();
+    }
+
+    @Test
+    @WithMockUser(roles = {"ADMIN"})
+    @Sql(scripts = "classpath:db/categories/remove-horror-from-category-table.sql",
+            executionPhase = Sql.ExecutionPhase.AFTER_TEST_METHOD)
+    @DisplayName("Create a new category with valid data")
+    void createCategory_validInput_success() throws Exception {
+        String expectedName = "Horror";
+        String expectedDescription =
+                "What unites the books in this genre "
+                        + "is not theme, plot, or setting, "
+                        + "but the feeling they inspire in the reader.";
+        CreateCategoryRequestDto requestDto = new CreateCategoryRequestDto();
+        requestDto.setName(expectedName);
+        requestDto.setDescription(expectedDescription);
+
+        mockMvc.perform(post(CATEGORY_ENDPOINT).contentType(MediaType.APPLICATION_JSON)
+                        .content(objectMapper.writeValueAsString(requestDto)))
+                .andExpect(status().isCreated()).andExpect(jsonPath("$.name").value(expectedName))
+                .andExpect(jsonPath("$.description").value(expectedDescription));
+    }
```

Comment:
I suggest to parse object received from  mockMvc.perform to business object and compare using asserEquals